source/frontend/src/features/mapSideBar/compensation/models/CompensationRequisitionFormModel.ts

```diff
@@ -115,18 +109,21 @@ export class CompensationRequisitionFormModel {
               } as ApiGen_Concepts_CompReqLeaseProperty;
             })
           : null,
-      compReqPayees: compReqPayees ?? [],
-      compReqLeaseStakeholders: isValidId(+this.leaseStakeholderId)
-        ? [
-            {
-              ...getEmptyBaseAudit(),
-              leaseStakeholderId: +this.leaseStakeholderId,
-              compensationRequisitionId: this.id,
-              compReqLeaseStakeholderId: null,
-              leaseStakeholder: null,
-            },
-          ]
-        : [],
+      compReqLeasePayees: this.payees
+        .filter(payee => payee.payeeType === PayeeType.LeaseStakeholder)
+        .map(payee => payee.toLeaseApi()),
+      compReqAcqPayees: this.payees
+        .filter(payee =>
+          [
+            PayeeType.AcquisitionTeam,
+            PayeeType.InterestHolder,
+            PayeeType.LegacyPayee,
+            PayeeType.Owner,
+            PayeeType.OwnerRepresentative,
+            PayeeType.OwnerSolicitor,
+          ].includes(payee.payeeType),
+        )
+        .map(payee => payee.toAcquisitionApi()),
```

Comment:
maybe this is an extensive check and we will never get a NPE but `payee.toAcquisitionApi()` and `payee.toLeaseApi()` can return NULL. See the old logic had an "exists" filter to not send nulls to the server.
I think we should have it here as well to protect against sending NULLs to the API